sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -94,7 +95,11 @@ object SparkStringUtils extends Logging {
       val restNum = seq.length - numFields
       val ending = (if (numFields == 0) "" else sep) +
         (if (restNum == 0) "" else s"... $restNum more fields") + end
-      seq.take(numFields).mkString(start, sep, ending)
+      if (customToString.isDefined) {
+        seq.take(numFields).map(customToString.get).mkString(start, sep, ending)
+      } else {
+        seq.take(numFields).mkString(start, sep, ending)
+      }
     } else {
       seq.mkString(start, sep, end)
```

Comment:
How about applying the custom converter here?

Reply:
Could you write a test for the case, please. Like top elements are less than maxFields, but some nested elem has more than maxFields.